spec/stack_factory_spec.rb

```diff
@@ -43,4 +44,37 @@
     })
   end
 
+  it 'errors if required tags are not present' do
+    allow(OpenStax::Aws.configuration).to receive(:required_stack_tags).and_return(%w(Application))
+
+    expect{
+      described_class.build(id: :bar, deployment: mock_deployment) do
+        name "something"
+        template_directory __dir__, "support/templates"
+        relative_template_path "simple.yml"
+        tag :foo, "bar"
+      end
+    }.to raise_error(/tag is required .* but is blank/)
+  end
+
+  it 'sets tags' do
+    stack = described_class.build(id: :bar, deployment: mock_deployment) do
+      name "something"
+      region 'us-east-1'
+      template_directory __dir__, "support/templates"
+      relative_template_path "simple.yml"
+      tag :foo, "bar"
+    end
+
+
+    expect(stack.tags).not_to be_empty
```

Comment:
separate tests (one for each "thing" being tested) would be nice.

Reply:
this one is really a way to say

`expect(stack.tags).to eq [{"foo": "bar"}]`

I just broke it out to not deal with object equality (`Tag`)... lame I know :-)